SonarRunner.Shim/SonarRunner.Wrapper.cs

```diff
@@ -67,25 +67,34 @@ public ProjectInfoAnalysisResult Execute(AnalysisConfig config, IEnumerable<stri
 
             ProjectInfoAnalysisResult result = PropertiesFileGenerator.GenerateFile(config, logger);
             Debug.Assert(result != null, "Not expecting the file generator to return null");
-
-            ProjectInfoReportBuilder.WriteSummaryReport(config, result, logger);
-
             result.RanToCompletion = false;
 
-            if (result.FullPropertiesFilePath == null)
-            {
-                // We expect a detailed error message to have been logged explaining
-                // why the properties file generation could not be performed
-                logger.LogInfo(Resources.MSG_PropertiesGenerationFailed);
-            }
-            else
-            {
-                string exeFileName = FindRunnerExe(config, logger);
-                if (exeFileName != null)
+            SonarProjectPropertiesValidator.Validate(
+                config.SonarRunnerWorkingDirectory, result.Projects,
+                onValid: () =>
```

Comment:
this is IODA design - to reduce coupling, I'm giving it a try... https://speakerdeck.com/jchapuis/ioda-architecture-for-loosely-coupled-programs